src/js/tech/chromecast.js

```diff
@@ -23,8 +23,19 @@ class Chromecast extends Tech {
         this.apiSession = this.options_.source.apiSession;
         this.receiver = this.apiSession.receiver.friendlyName;
 
-        this.apiMedia.addUpdateListener(::this.onMediaStatusUpdate);
-        this.apiSession.addUpdateListener(::this.onSessionUpdate);
+        let mediaStatusUpdateHandler = ::this.onMediaStatusUpdate;
+        let sessionUpdateHanlder = ::this.onSessionUpdate;
+
+        this.apiMedia.addUpdateListener(mediaStatusUpdateHandler);
+        this.apiSession.addUpdateListener(sessionUpdateHanlder);
+
+        this.on('dispose', () => {
```

Comment:
I had originally done that, but this was using the es6 `::this.onSessionUpdate`, which just binds the method to `this`, which returns a new instance of the method. So, in the `dispose` method, there was no access to the new instance of the bound method:

```
constructor() {
  this.apiMedia.addUpdateLister(::this.onSessionUpdate);
}

dispose() {
  // Because the `::` shorthand returns a new instance, the listener is not removed
  this.apiMedia.removeUpdateListener(::this.onSessionUpdate);
}
```

My first pass of changes was:
```
constructor() {
  this.onSessionUpdate = this.onSessionUpdate.bind(this);

  this.apiMedia.addUpdateLister(this.onSessionUpdate);
}

dispose() {
  this.apiMedia.removeUpdateListener(this.onSessionUpdate);
}
```

I didn't think it flowed with the way the other handlers in the constructor were setup, so I just opted to follow the precedent.

/shrug

😄